apps/studio/src/components/export/ExportManager.vue

```diff
@@ -59,48 +63,71 @@ export default Vue.extend({
     return {
       // these are like 'pending Export'
       table: (undefined as TableOrView | undefined),
+      query: '',
+      query_name: '',
       filters: (undefined as TableFilter[] | undefined),
     }
   },
   computed: {
     ...mapGetters({ 'exports': 'exports/runningExports' }),
     rootBindings(): RootBinding[] {
       return [
-        { event: AppEvent.beginExport, handler: this.handleExportRequest }
+        { event: AppEvent.beginExport, handler: this.handleExportRequest },
       ]
     }
   },
   methods: {
     ...mapMutations({ addExport: "exports/addExport" }),
     async startExport(options: StartExportOptions) {
-        const exporter = new ExportClassPicker[options.exporter](
-          options.filePath,
-          this.connection,
-          options.table,
-          options.filters || [],
-          options.options,
-          options.outputOptions
-        )
-        this.addExport(exporter)
-        exporter.onProgress(this.notifyProgress.bind(this))
-        await exporter.exportToFile()
-        if (exporter.status !== ExportStatus.Completed) return;
-        const n = this.$noty.success(`Export of ${options.table.name} complete`, {
+      const exporter = new ExportClassPicker[options.exporter](
+        options.filePath,
+        this.connection,
+        options.table,
+        options.query,
+        options.query_name,
+        options.filters || [],
+        options.options,
+        options.outputOptions
+      )
+      this.addExport(exporter)
+      exporter.onProgress(this.notifyProgress.bind(this))
+
+      const exportName = options.table? options.table.name : options.query_name;
+
+      await exporter.exportToFile()
+
+      if (exporter.status == ExportStatus.Error) {
+        const exportName = options.table? options.table.name : options.query_name;
+        const error_notice = this.$noty.error(`Export of ${exportName} failed: ${exporter.error}`, {
           buttons: [
-            Noty.button('Show', "btn btn-primary", () => {
-              this.$native.files.showItemInFolder(options.filePath)
-              n.close()
+            Noty.button('Close', "btn btn-primary", () => {
+              error_notice.close()
             })
           ]
-        })
+        }).setTimeout(60000) // 1 minute timeout
+        return
+      }
+      if (exporter.status !== ExportStatus.Completed) return;
+      const n = this.$noty.success(`Export of ${exportName} complete`, {
```

Comment:
not sure where it is, but if you cancel an export it says `undefined export cancelled`

Reply:
Fixed in 980017d